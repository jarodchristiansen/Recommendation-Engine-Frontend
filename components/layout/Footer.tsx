import Link from "next/link";

const Footer = () => {
  const currentYear = new Date().getFullYear();

  const links = [
    { id: 1, text: "home", link: "/" },
    { id: 2, text: "dashboard", link: "/dashboard" },
    { id: 3, text: "recommendations", link: "/recommendations" },
    // { id: 4, text: "about", link: "/about" },
    // { id: 5, text: "contact", link: "/contact" },
    { id: 6, text: "Sign In/Up", link: "/auth" },
  ];

  return (
    <footer className="bg-black text-white py-8">
      <div className="container mx-auto ">
        {/* Main Navigation Links */}
        <div className="flex flex-col items-center justify-center text-center gap-6 md:flex-row md:gap-8 border-b border-gray-700 pb-4 mb-4">
          {links.map(({ id, text, link }) => (
            <Link href={link} key={id}>
              <span className="text-gray-400 hover:text-white px-3 py-2 capitalize transition duration-300 ease-in-out">
                {text}
              </span>
            </Link>
          ))}
        </div>

        {/* Copyright Info */}
        <div className="text-center text-gray-400">
          <p>&copy; {currentYear} All rights reserved.</p>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
