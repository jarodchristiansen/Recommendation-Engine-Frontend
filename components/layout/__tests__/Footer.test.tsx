import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Footer from "@/components/layout/Footer"; // Adjust path as necessary

describe("Footer Component", () => {
  beforeEach(() => {
    render(<Footer />);
  });

  // snapshhot test
  it("matches the snapshot of the Footer", () => {
    const { container } = render(<Footer />);

    // Create a snapshot of the rendered Footer
    expect(container).toMatchSnapshot();
  });

  it("displays the current year", () => {
    const currentYear = new Date().getFullYear();
    expect(
      screen.getByText(`Â© ${currentYear} All rights reserved.`)
    ).toBeInTheDocument();
  });
});
