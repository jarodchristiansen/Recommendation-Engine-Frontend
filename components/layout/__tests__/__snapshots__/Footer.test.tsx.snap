// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches the snapshot of the Footer 1`] = `
<div>
  <footer
    class="bg-black text-white py-8"
  >
    <div
      class="container mx-auto "
    >
      <div
        class="flex flex-col items-center justify-center text-center gap-6 md:flex-row md:gap-8 border-b border-gray-700 pb-4 mb-4"
      >
        <a
          href="/"
        >
          <span
            class="text-gray-400 hover:text-white px-3 py-2 capitalize transition duration-300 ease-in-out"
          >
            home
          </span>
        </a>
        <a
          href="/dashboard"
        >
          <span
            class="text-gray-400 hover:text-white px-3 py-2 capitalize transition duration-300 ease-in-out"
          >
            dashboard
          </span>
        </a>
        <a
          href="/recommendations"
        >
          <span
            class="text-gray-400 hover:text-white px-3 py-2 capitalize transition duration-300 ease-in-out"
          >
            recommendations
          </span>
        </a>
        <a
          href="/auth"
        >
          <span
            class="text-gray-400 hover:text-white px-3 py-2 capitalize transition duration-300 ease-in-out"
          >
            Sign In/Up
          </span>
        </a>
      </div>
      <div
        class="text-center text-gray-400"
      >
        <p>
          Â© 
          2024
           All rights reserved.
        </p>
      </div>
    </div>
  </footer>
</div>
`;
